compiler/rmi/rmi-interface/src/org/jetbrains/kotlin/rmi/DaemonParams.kt

```diff
@@ -0,0 +1,313 @@
+//    open fun extract(prefix: String = COMPILE_DAEMON_CMDLINE_OPTIONS_PREFIX): List<String> =
+//            when {
+//                skipIf(prop.get(dest)) -> listOf<String>()
+//                mergeWithDelimiter != null -> listOf(prefix + name + mergeWithDelimiter + convert(prop.get(dest))).filterNotNull()
+//                else -> listOf(prefix + name, convert(prop.get(dest))).filterNotNull()
+//            }
+//}
+//
+//class BoolPropExtractor<C, P: KMutableProperty1<C, Boolean>>(dest: C, prop: P, name: String? = null)
+//    : PropExtractor<C, Boolean, P>(dest, prop, name ?: prop.name, convert = { null }, skipIf = { !prop.get(dest) })
+//
+//class RestPropExtractor<C, P: KMutableProperty1<C, out MutableCollection<String>>>(dest: C, prop: P) : PropExtractor<C, MutableCollection<String>, P>(dest, prop, convert = { null }) {
+//    override fun extract(prefix: String): List<String> = prop.get(dest).map { prefix + it }
+//}
+//
+//
+//open class PropParser<C, V, P: KMutableProperty1<C, V>>(val dest: C,
+//                                                        val prop: P, alternativeNames: List<String>,
+//                                                        val parse: (s: String) -> V,
+//                                                        val allowMergedArg: Boolean = false) {
+//    val names = listOf(prop.name) + alternativeNames
+//    constructor(dest: C, prop: P, parse: (s: String) -> V, allowMergedArg: Boolean = false) : this(dest, prop, listOf(), parse, allowMergedArg)
+//    fun apply(s: String) = prop.set(dest, parse(s))
+//}
+//
+//class BoolPropParser<C, P: KMutableProperty1<C, Boolean>>(dest: C, prop: P): PropParser<C, Boolean, P>(dest, prop, { true })
+//
+//class RestPropParser<C, P: KMutableProperty1<C, MutableCollection<String>>>(dest: C, prop: P): PropParser<C, MutableCollection<String>, P>(dest, prop, { arrayListOf() }) {
+//    fun add(s: String) { prop.get(dest).add(s) }
+//}
+
+// --------------------------------------------------------
+
+open class PropMapper<C, V, P: KMutableProperty1<C, V>>(val dest: C,
+                                                        val prop: P,
+                                                        val names: List<String> = listOf(prop.name),
+                                                        val fromString: (s: String) -> V,
+                                                        val toString: ((v: V) -> String?) = { it.toString() },
+                                                        val skipIf: ((v: V) -> Boolean) = { false },
+                                                        val mergeDelimiter: String? = null)
+{
+    open fun toArgs(prefix: String = COMPILE_DAEMON_CMDLINE_OPTIONS_PREFIX): List<String> =
+            when {
+                skipIf(prop.get(dest)) -> listOf<String>()
+                mergeDelimiter != null -> listOf(prefix + names.first() + mergeDelimiter + toString(prop.get(dest))).filterNotNull()
```

Comment:
`mergeDelimiter` and the result of `toString` can be null.
Try to use `String?.orEmpty()` to process this case correctly